### ISSUE TITLE
Complex batch normalization

### ISSUE THREAD

**Issue by melsophos** [google/flax#2502]
I would be interested to see an implementation of batch normalization for complex neural networks, as defined in section 3.5 of [arXiv:1705.09792](https://arxiv.org/abs/1705.09792).

The paper argues that the usual batch normalization does not work for complex numbers. Naive use in Flax confirms this, since parameters are of type float32 (and I get various bugs which prevented me to make it work).

**Comment by jheek** [comment]
By default parameters are in float32 you can add a complex shift & scale by using `param_dtype=jnp.complex64`.
Perhaps you could elaborate what other issues you ran into?

**Comment by melsophos** [comment]
Sorry for not getting back to you sooner, I am very busy now and did not find time to make an appropriate explanation. Let me come back to you later.

**Comment by marcvanzee** [comment]
@melsophos sounds good, for now I am closing this issue since it is open for 2 months and it is not clear what the problem is, but feel free to re-open it once you have a more detailed explanation.
